.github/workflows/examples.yml

```diff
@@ -101,15 +100,12 @@ jobs:
 
       - name: Check diff
         id: check-diff
+        if: github.event_name == 'pull_request'
         run: |
-          if [ ! "$(git branch --show-current)" == "master" ]; then
-            git fetch origin master:master
-          fi
-
-          REGEXP="Dockerfile\|\.dockerignore"
-          CHANGED_FILES=$(git diff --name-only master..HEAD | grep "$REGEXP") || true;
+          REPO="${{ github.repository }}"
+          PR_NUMBER="${{ github.event.pull_request.number }}"
+          CHANGED_FILES=$(python dev/list_changed_files.py --repository $REPO --pr-num $PR_NUMBER | grep "Dockerfile\|\.dockerignore" || true);
           DOCKER_CHANGED=$([[ ! -z "$CHANGED_FILES" ]] && echo "true" || echo "false")
-
```

Comment:
To skip running this job on a feature branch, just check changed files in a PR, instead of diff between master and the PR branch.